src/TrackerEvents/BuyUnitsEvent.cpp

```diff
@@ -0,0 +1,23 @@
+// Copyright 2021 the Drakhtar authors. All rights reserved. MIT license.
+
+#include "TrackerEvents/BuyUnitsEvent.h"
+
+BuyUnitsEvent::BuyUnitsEvent(std::vector<UnitPurchase> unitsPurchased)
+    : TrackerEvent(UNITS_PURCHASED), unitsPurchased_(unitsPurchased) {}
+
+std::string BuyUnitsEvent::toJson() {
+  std::string str = ",\n";
+  str += "    {\n";
+  str += R"(      "Event Type": "Buy Units Event",)";
+  str += "\n" + TrackerEvent::toJson() + +",\n";
+
+  str += "    (      \"Units purchased#\": )\n";
+  for (int i = 0; i < unitsPurchased_.size(); i++) {
```

Comment:
Para solucionar la alerta `-Wsign-compare` del build:
```suggestion
  for (size_t i = 0; i < unitsPurchased_.size(); i++) {
```